File: code_snippets/ts/src/concepts/invocations/one_way.ts
import * as restate from "@restatedev/restate-sdk";
import { greetCounter, greeter, myWorkflow } from "./utils";

// <start_one_way_call>
async function myRestateHandler(ctx: restate.Context) {
  // !focus(1:2)
  // From a Restate handler
  ctx.serviceSendClient(greeter).greet({ greeting: "Hi" });
}
// <end_one_way_call>

File: code_snippets/python/src/concepts/invocations/one_way.py
from restate import Service, Context

from src.concepts.invocations.utils import greet

my_service = Service("MyService")


# <start_one_way_call>
@my_service.handler()
async def my_handler(ctx: Context, arg):
    # focus
    ctx.service_send(greet, arg="Hi")
# <end_one_way_call>
